api/v1alpha1/olsconfig_types.go

```diff
@@ -94,6 +94,26 @@ type OLSSpec struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Introspection Enabled",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
 	IntrospectionEnabled bool `json:"introspectionEnabled,omitempty"`
+	// RAG databases
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="RAG Databases",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
+	RAG []RAGSpec `json:"rag,omitempty"`
+}
+
+// RAGSpec defines how to retrieve a RAG databases.
+type RAGSpec struct {
+	// +kubebuilder:validation:Required
+	// +required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Index Path in the Image"
+	IndexPath string `json:"indexPath,omitempty"`
+	// +kubebuilder:validation:Required
+	// +required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Index ID"
+	IndexID string `json:"indexID,omitempty"`
```

Comment:
@raptorsun How does the user name this IndexID ?

Reply:
In the storage there can be multiple indexes, the user should tell OLS to load which index in the path.
This IndexID correspond to the index_id here: https://github.com/openshift/lightspeed-service/blob/main/ols/src/rag_index/index_loader.py#L107

Reply:
@rh-tokeefe - we need to mention this in our doc